### PR TITLE
Change plots to use plotnine

### PR DESCRIPTION
**Pull request by patrick-miller**
Here I am replacing all of the plots with the ggplot inspired `plotnine` package. I think that these look good to me, but let me know what you think.

### CODE REVIEW

**Comment by rdvelazquez** [comment]
Nice work. These look really good based on my first look at them. A few quick comments:

1. Should you also revise the utils.py file to remove the functions that we no longer need?

2. I really like the revision to the probabilities plots at the end of the notebook (showing the plots for all three models). I think labeling the axis would also be very helpful.

File: scripts/2.mutation-classifier.py
Comment (dhimmel) [inline scripts/2.mutation-classifier.py]: Let's avoid `import *` ([reasons](https://stackoverflow.com/a/2386740/4651668)). If you're worried about plotnine taking up lot's of space you could do something like `import plotnine as pln` or `import plotnine as gg`.

File: scripts/2.mutation-classifier.py
Comment (dhimmel) [inline scripts/2.mutation-classifier.py]: I'm not sure a barplot is the best. I'm thinking a scatterplot with model shown via line/dot color or via facet.

You could do a [`scale_x_log10`](http://plotnine.readthedocs.io/en/stable/generated/plotnine.scales.scale_x_log10.html) so the x-axis is continuous rather than discreet.

Ideally plot would expand_limits for y to include 0.5 and 1. This will have to wait until the next plotnine release: see https://github.com/has2k1/plotnine/commit/192609d761726c84ffec38dcbade325d9556364e. In other words, 0.5-1.0 should be the ylim unless AUROCs are below 0.5 in which case the lower bound should accommodate the lowest AUROC.

File: scripts/2.mutation-classifier.py
Comment (dhimmel) [inline scripts/2.mutation-classifier.py]: Minor: models as rows and partition as columns makes a little more sense to me.

**Review by dhimmel — COMMENTED** [review]
Nice. What do you think about the default plotnine style? In general, I find `theme_bw` to be more aesthetic. If we want, we can define a style in utils. Always can do this in a future PR. Here are some [ggplot2 mods](https://github.com/dhimmel/hetior/blob/e24db93e99fb75085d6b186ecf052cba516c05e0/R/plotting.R) I often use.

File: scripts/2.mutation-classifier.py
Comment (patrick-miller) [reply scripts/2.mutation-classifier.py]: Agreed, I was just copying from some `plotnine` examples.

File: scripts/2.mutation-classifier.py
Comment (patrick-miller) [reply scripts/2.mutation-classifier.py]: I switched to a scatterplot and the log scale -- it initially looked funky to me with the default theme. I used a hack to get 'expand limits'.

File: scripts/2.mutation-classifier.py
Comment (patrick-miller) [reply scripts/2.mutation-classifier.py]: Done

**Comment by patrick-miller** [comment]
I translated one of my R ggplot themes to `plotnine` in the utils.py file. We can play around with this and the color scheme for the plots. The ROC plot might be a little busy.

**Comment by rdvelazquez** [comment]
Would it be possible to show the actual AUROC values on the AUROC plot? Maybe next to the lines on the legend? If not maybe we can insert a table that shows this. I think this is one of the more important pieces of information in the notebook so it would be good to be able to easily see the testing and training AUROC for each model.

**Review by dhimmel — APPROVED** [review]

**Comment by dhimmel** [comment]
@rdvelazquez I gave you write access to this repo. After all your suggestions have been addressed and @patrick-miller is done, you can "Squash and merge" this pull request. Make sure to select "Squash and merge" rather than "Create a merge commit"

**Comment by patrick-miller** [comment]
I agree @rdvelazquez about the AUROC values. I include them in the next few cells, but it would be better to have them inside the plot itself. Because there are 6 values, things can get cluttered. Maybe we only include the test AUROC inside the plot?

**Comment by dhimmel** [comment]
I'm fine with the current display of AUROCs (in the following cell). I don't think there is really a clean way to include them in the plot.

File: utils.py
Comment (dhimmel) [inline utils.py]: Consider changing `#e5e5e5` to `#FEF2E2` for `strip.background` color to make the facet strips a little less drab.

Comment (patrick-miller) [reply utils.py]: Will do

**Comment by dhimmel** [comment]
@patrick-miller merge when you're ready. So ml-workers should be able now to directly use this notebook?

@dcgoss how should we go about using this notebook in production? Can ml-workers just use the URL for this notebook once it's merged?

**Comment by rdvelazquez** [comment]
> how should we go about using this notebook in production? Can ml-workers just use the URL for this notebook once it's merged?

We are planning to use the machine-learning repo for development of the notebooks but ml-workers for production so the next step (either now or once all the items in #110 are done) will be to open a PR in ml-workers updating that notebook. @patrick-miller 's revisions in #111 were designed to let this notebook work in both the machine-learning and the ml-workers repos. 

I'm not opposed to just having ml-workers reference this notebook but I think when we discussed it at the last meetup there were some potential issues with doing that.

**Pull request merged**
